lib/infer.cpp

```diff
@@ -388,3 +387,35 @@ std::vector<MathLib::bigint> getMaxValue(const ValuePtr<InferModel>& model, cons
         return model->match(v);
     }).maxvalue;
 }
+
+namespace {
+    struct IntegralInferModel : InferModel {
```

Comment:
Maybe the `SymbolicInferModel` should be moved here as well.

Reply:
Makes sense. Will do in a follow-up.

Reply:
It makes sense to put them all in one place. But keeping them in the only place that uses them might also make sense. The compiler would also be able to see all the code...not 100% on what to do here.

There is also `IteratorInferModel` and related classes in `vf_infercondition.cpp`.